src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Completion/TagHelperCompletionProvider.cs

```diff
@@ -60,6 +60,12 @@ public ImmutableArray<RazorCompletionItem> GetCompletionItems(RazorCompletionCon
             _ => owner.Parent
         };
 
+        if (owner is null)
+        {
+            Debug.Assert(context.Owner is RazorDocumentSyntax, $"Expecting null parent only on {nameof(RazorDocumentSyntax)}");
+            return [];
```

Comment:
Rather than return nothing for RazorDocumentSyntax, can we not make it another special case in the switch above, where we don't get the Parent since we know it will be null?

Reply:
Yep, that works, though we still return nothing downstream. See next commit.